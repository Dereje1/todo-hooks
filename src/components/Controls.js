import IconButton from '@mui/material/IconButton';
import ButtonGroup from '@mui/material/ButtonGroup';
import AddIcon from '@mui/icons-material/Add';
import CheckIcon from '@mui/icons-material/Check';
import AllInclusiveIcon from '@mui/icons-material/AllInclusive';
import NotificationImportantIcon from '@mui/icons-material/NotificationImportant';
import Tooltip from '@mui/material/Tooltip';

const Controls = ({
    setOpenTaskDialog,
    setFilterSetting,
    filterSetting
}) =>
(
    <div id="button-group-container"
        style={{
            display: 'flex',
            justifyContent: 'center',
            marginTop: 10,
            marginBottom: 10,
        }}
    >
        <ButtonGroup variant="contained" aria-label="outlined primary button group" sx={{ padding: 1 }}>
            <Tooltip title="Add a task">
                <IconButton
                    id="add-task"
                    onClick={setOpenTaskDialog}
                    sx={{ margin: 1 }}
                >
                    <AddIcon />
                </IconButton>
            </Tooltip>

            <Tooltip title="Completed tasks">
                <IconButton
                    id="completed-tasks"
                    onClick={() => setFilterSetting('completed')}
                    sx={{ border: filterSetting === 'completed' ? '.5px solid black' : '', margin: 1 }}
                >
                    <CheckIcon color='primary' />
                </IconButton>
            </Tooltip>

            <Tooltip title="Active tasks">
                <IconButton
                    id="active-tasks"
                    onClick={() => setFilterSetting('active')}
                    sx={{ border: filterSetting === 'active' ? '.5px solid black' : '', margin: 1 }}
                >
                    <NotificationImportantIcon color='error' />
                </IconButton>
            </Tooltip>

            <Tooltip title="All tasks">
                <IconButton
                    id="all-tasks"
                    onClick={() => setFilterSetting('all')}
                    sx={{ border: filterSetting === 'all' ? '.5px solid black' : '', margin: 1 }}
                >
                    <AllInclusiveIcon color='success' />
                </IconButton>
            </Tooltip>
        </ButtonGroup>
    </div>
)

export default Controls;