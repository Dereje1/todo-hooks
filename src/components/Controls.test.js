import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Controls from "./Controls";

const props = {
    setOpenTaskDialog: jest.fn(),
    setFilterSetting: jest.fn(),
    filterSetting: 'all'
}

test('will render', () => {
    const wrapper = shallow(<Controls {...props} />)
    expect(toJson(wrapper)).toMatchSnapshot();
})

test('will set the filter status to complete',()=>{
    const updatedProps = {...props, filterSetting: 'completed'}
    const wrapper = shallow(<Controls {...updatedProps} />)
    let completedButton = wrapper.find({id: 'completed-tasks'})
    completedButton.props().onClick();
    expect(props.setFilterSetting).toHaveBeenCalledWith('completed')
    expect(completedButton.props().sx.border).toBe('.5px solid black')
})

test('will set the filter status to active',()=>{
    const updatedProps = {...props, filterSetting: 'active'}
    const wrapper = shallow(<Controls {...updatedProps} />)
    const activeButton = wrapper.find({id: 'active-tasks'})
    activeButton.props().onClick();
    expect(props.setFilterSetting).toHaveBeenCalledWith('active')
    expect(activeButton.props().sx.border).toBe('.5px solid black')
})

test('will set the filter status to all',()=>{
    const updatedProps = {...props, filterSetting: 'all'}
    const wrapper = shallow(<Controls {...updatedProps} />)
    const allButton = wrapper.find({id: 'all-tasks'})
    allButton.props().onClick();
    expect(props.setFilterSetting).toHaveBeenCalledWith('all')
    expect(allButton.props().sx.border).toBe('.5px solid black')
})