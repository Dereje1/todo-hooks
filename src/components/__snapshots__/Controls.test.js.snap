// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`will render 1`] = `
<div
  id="button-group-container"
  style={
    Object {
      "display": "flex",
      "justifyContent": "center",
      "marginBottom": 10,
      "marginTop": 10,
    }
  }
>
  <ForwardRef(ButtonGroup)
    aria-label="outlined primary button group"
    sx={
      Object {
        "padding": 1,
      }
    }
    variant="contained"
  >
    <ForwardRef(Tooltip)
      title="Add a task"
    >
      <ForwardRef(IconButton)
        id="add-task"
        onClick={[MockFunction]}
        sx={
          Object {
            "margin": 1,
          }
        }
      >
        <Memo(ForwardRef(AddIcon)) />
      </ForwardRef(IconButton)>
    </ForwardRef(Tooltip)>
    <ForwardRef(Tooltip)
      title="Completed tasks"
    >
      <ForwardRef(IconButton)
        id="completed-tasks"
        onClick={[Function]}
        sx={
          Object {
            "background": "",
            "margin": 1,
          }
        }
      >
        <Memo(ForwardRef(CheckIcon))
          color="primary"
        />
      </ForwardRef(IconButton)>
    </ForwardRef(Tooltip)>
    <ForwardRef(Tooltip)
      title="Active tasks"
    >
      <ForwardRef(IconButton)
        id="active-tasks"
        onClick={[Function]}
        sx={
          Object {
            "background": "",
            "margin": 1,
          }
        }
      >
        <Memo(ForwardRef(NotificationImportantIcon))
          color="error"
        />
      </ForwardRef(IconButton)>
    </ForwardRef(Tooltip)>
    <ForwardRef(Tooltip)
      title="All tasks"
    >
      <ForwardRef(IconButton)
        id="all-tasks"
        onClick={[Function]}
        sx={
          Object {
            "background": "#e8e6e6",
            "margin": 1,
          }
        }
      >
        <Memo(ForwardRef(AllInclusiveIcon))
          color="success"
        />
      </ForwardRef(IconButton)>
    </ForwardRef(Tooltip)>
  </ForwardRef(ButtonGroup)>
</div>
`;
